data-science-onramp/data-ingestion/setup_test.py

```diff
@@ -0,0 +1,181 @@
+import os
+import re
+
+import uuid
+
+from google.cloud import dataproc_v1 as dataproc
+from google.cloud import storage
+
+import pytest
+
+
+# Set global variables
+PROJECT = os.environ['GCLOUD_PROJECT']
+REGION = "us-central1"
+ZONE = "us-central1-a"
+CLUSTER_NAME = f'setup-test-{uuid.uuid4()}'
+BUCKET_NAME = f'setup-test-code-{uuid.uuid4()}'
+
+BUCKET = None
+
+
+@pytest.fixture(autouse=True)
+def setup_and_teardown_cluster():
+    # Create cluster configuration
+    zone_uri = \
+        f'https://www.googleapis.com/compute/v1/projects/{PROJECT}/zones/{ZONE}'
+    cluster_data = {
+        'project_id': PROJECT,
+        'cluster_name': CLUSTER_NAME,
+        'config': {
+            'gce_cluster_config': {
+                'zone_uri': zone_uri,
+                "metadata": {
+                    "PIP_PACKAGES": "google-cloud-storage"
+                },
+            },
+            'master_config': {
+                'num_instances': 1,
+                'machine_type_uri': 'n1-standard-8'
+            },
+            'worker_config': {
+                'num_instances': 6,
+                'machine_type_uri': 'n1-standard-8'
+            },
+            "initialization_actions": [
+                {
+                    "executable_file": ("gs://dataproc-initialization-actions/"
+                                        "python/pip-install.sh"),
+                }
+            ],
+            "software_config": {
+                "image_version": "1.5.4-debian10",
+                "optional_components": [
+                    "ANACONDA"
+                ],
+            }
+        }
+    }
+
+    # Create cluster using cluster client
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': '{}-dataproc.googleapis.com:443'.format(REGION)
+    })
+
+    operation = cluster_client.create_cluster(PROJECT, REGION, cluster_data)
+
+    # Wait for cluster to provision
+    operation.result()
+
+    yield
+
+    # Delete cluster
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': f'{REGION}-dataproc.googleapis.com:443'
+    })
+
+    operation = cluster_client.delete_cluster(PROJECT, REGION,
+                                              CLUSTER_NAME)
+    operation.result()
+
+
+@pytest.fixture(autouse=True)
+def setup_and_teardown_bucket():
+    global BUCKET
+    # Create GCS Bucket
+    storage_client = storage.Client()
+    BUCKET = storage_client.create_bucket(BUCKET_NAME)
+
+    yield
```

Comment:
instead of using a global variable `BUCKET`, you can yield the bucket itself, then pass the fixture to the test as an argument. So, in line 87 you'd have
```
bucket = storage_client.create_bucket(BUCKET_NAME)
yield bucket
```

then in the test itself, you'd call 
`def test_setup(capsys,setup_and_teardown_bucket )`

and replace every other instance of `BUCKET` with `setup_and_teardown_bucket`. 

That said, it might be worth renaming the fixture to `test_bucket` or something similar that you like, since that's what you're yielding.